### PR TITLE
added requiredBinarayComparators

### PR DESCRIPTION
**Pull request by MansurAshraf**

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/serialization/RequiredBinaryComparators.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  implicit def ordSer[T]: OrderedSerialization[T] = macro com.twitter.scalding.serialization.macros.impl.OrderedSerializationProviderImpl[T]
+
+  override def config =
+    super.config + (Config.ScaldingRequireOrderedSerialization -> "true")
+}
+
+object RequiredBinaryComparators {
+
+  implicit def ordSer[T]: OrderedSerialization[T] = macro com.twitter.scalding.serialization.macros.impl.OrderedSerializationProviderImpl[T]
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/serialization/RequiredBinaryComparators.scala]: can we change the name of this to say, `orderedSerialization`? Scala has this issue (bug?) where if you have two implicits in scope of the same name you get issues:
```
scala> object Test { implicit val name: String = "object" }
defined module Test

scala> trait TestT { implicit val name: String = "trait" }
defined trait TestT

scala> class TestClass extends TestT { import Test._; println(implicitly[String]) }
<console>:9: error: could not find implicit value for parameter e: String
       class TestClass extends TestT { import Test._; println(implicitly[String]) }
                                                                        ^

scala> import Test._
import Test._

scala> class TestClass extends TestT { println(implicitly[String]) }
defined class TestClass

scala> new TestClass
trait
res0: TestClass = TestClass@20c9cd89

scala> class TestClass extends TestT { import Test._; println(implicitly[String]) }
<console>:12: error: could not find implicit value for parameter e: String
       class TestClass extends TestT { import Test._; println(implicitly[String]) }
                                                                        ^

scala> object Test2 { val name2: String = "Test2" }
defined module Test2

scala> class TestClass extends TestT { import Test2._; println(implicitly[String]) }
defined class TestClass

scala> new TestClass
trait
res1: TestClass = TestClass@3ba4236f
```

**Comment by johnynek** [comment]
merge when green

**Comment by MansurAshraf** [comment]
closing this and moving the change to https://github.com/twitter/scalding/pull/1338 in order to use thrift macro

**Comment by johnynek** [comment]
PS: you wasted the 1337 PR. :( you could have been so leet!

**Pull request closed without merging**